src/components/course/data/hooks.jsx

```diff
@@ -598,7 +597,7 @@ export const useCheckSubsidyAccessPolicyRedeemability = ({
   isQueryEnabled,
   queryOptions,
 }) => {
-  const { id: lmsUserId } = getAuthenticatedUser();
+  const { authenticatedUser: { id: lmsUserId } } = useContext(AppContext);
```

Comment:
[nit/suggestion] It looks like `authenticatedUser` contains both `id` and `userId` representing the LMS user's identifier. It appears that we're a bit inconsistent in which we use. It may be good to ensure that we always rely on `userId` or `id` for consistency, but not both in various usages. FWIW, not sure why `@edx/frontend-platform` returns 2 duplicate ID fields like this.